x-pack/solutions/search/plugins/search_playground/public/components/saved_playground/delete_playground_modal.tsx

```diff
@@ -0,0 +1,110 @@
+  playgroundId: string;
+  playgroundName: string;
+  onClose: () => void;
+}
+
+export const DeletePlaygroundModal = ({
+  playgroundId,
+  playgroundName,
+  onClose,
+}: DeletePlaygroundModalProps) => {
+  const { application, notifications } = useKibana().services;
+  const { deleteSavedPlayground, isLoading } = useDeleteSavedPlayground();
+  const onDeleteSuccess = useCallback(() => {
+    onClose();
+    application.navigateToApp(PLUGIN_ID);
+  }, [application, onClose]);
+  const onDeleteError = useCallback(
+    (error: unknown) => {
+      onClose();
+      const errorMessage = getErrorMessage(error);
+      notifications.toasts.addError(error instanceof Error ? error : new Error(errorMessage), {
+        title: i18n.translate('xpack.searchPlayground.savedPlayground.deleteError.title', {
+          defaultMessage: 'Error deleting playground',
+        }),
+        toastMessage: errorMessage,
+      });
+    },
+    [notifications.toasts, onClose]
+  );
+  const onDelete = useCallback(
+    (e: React.MouseEvent<HTMLButtonElement, MouseEvent>) => {
+      e.preventDefault();
+
+      deleteSavedPlayground(playgroundId, {
+        onSuccess: onDeleteSuccess,
+        onError: onDeleteError,
+      });
+    },
+    [deleteSavedPlayground, onDeleteError, onDeleteSuccess, playgroundId]
+  );
+  const modalTitleId = useGeneratedHtmlId();
+
+  return (
+    <EuiConfirmModal
+      data-test-subj="deletePlaygroundActionModal"
+      aria-labelledby={modalTitleId}
+      title={i18n.translate(
+        'xpack.searchPlayground.savedPlayground.deletePlayground.confirmModal.modalTitle',
+        {
+          defaultMessage: 'Delete playground',
+        }
+      )}
+      titleProps={{ id: modalTitleId }}
+      onCancel={onClose}
+      onConfirm={onDelete}
+      isLoading={isLoading}
+      buttonColor="danger"
+      cancelButtonText={i18n.translate(
+        'xpack.searchPlayground.savedPlayground.deletePlayground.confirmModal.cancelButtonText',
+        {
+          defaultMessage: 'Cancel',
+        }
+      )}
+      confirmButtonText={i18n.translate(
+        'xpack.searchPlayground.savedPlayground.deletePlayground.confirmModal.confirmButtonText',
+        {
+          defaultMessage: 'Delete playground',
+        }
+      )}
+    >
+      <>
+        <EuiText>
+          <p>
+            <FormattedMessage
+              id="xpack.searchPlayground.savedPlayground.deletePlayground.deleteDescription"
+              defaultMessage="You are about to delete the {playgroundName} playground"
+              values={{
+                playgroundName: <strong>{playgroundName}</strong>,
+              }}
+            />
+          </p>
+          <p>
+            <FormattedMessage
+              id="xpack.searchPlayground.savedPlayground.deletePlayground.deleteWarningDescription"
+              defaultMessage="You can't recover a deleted playground. Make sure you have appropriate backups."
+            />
+          </p>
+        </EuiText>
```

Comment:
@leemthompo This copy was 🍝  from another delete modal so happy to adjust if needed.